expression/builtin_miscellaneous.go

```diff
@@ -295,7 +295,53 @@ type builtinIsIPv4Sig struct {
 
 // See https://dev.mysql.com/doc/refman/5.7/en/miscellaneous-functions.html#function_is-ipv4
 func (b *builtinIsIPv4Sig) eval(row []types.Datum) (d types.Datum, err error) {
-	return d, errFunctionNotExists.GenByArgs("IS_IPV4")
+	args, err := b.evalArgs(row)
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	if args[0].IsNull() {
+		d.SetInt64(0)
+		return d, nil
+	}
+	// isIPv4(str)
+	// args[0] string
+	s, err := args[0].ToString()
+	if err != nil {
+		return d, errors.Trace(err)
+	}
+	if isIPv4(s) {
+		d.SetInt64(1)
+	} else {
+		d.SetInt64(0)
+	}
+	return d, nil
+}
+
+// isIPv4 checks IPv4 address which satisfying the format A.B.C.D(0<=A/B/C/D<=255).
+// mapped IPv6 address like '::ffff:1.2.3.4' would return false
+func isIPv4(ip string) bool {
+	// acc: keep the decimal value of each segment under check. should between 0 and 255 for valid IPv4 address.
```

Comment:
keep the decimal value of each segment under check**, which should be** between 0 and 255 for valid IPv4 address.